### PR TITLE
Improve AQA_Test_Pipeline for release testing

### PR DESCRIPTION
**Pull request by llxia**
Initial version for release testing improvement.
If PLATFORMS is set to release, the *.json file is used to trigger release testing.

related: https://github.com/adoptium/aqa-tests/issues/5510
fixes: https://github.com/adoptium/aqa-tests/issues/5509
fixes: https://github.com/eclipse-openj9/openj9/issues/20004

### CODE REVIEW

**Comment by llxia** [comment]
[Internal testing](https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/AQA_Test_Pipeline_Release/61/)

**Review by LongyuZhang — APPROVED** [review]
LGTM

File: buildenv/jenkins/aqaTestPipeline.groovy [hunk 1]
```diff
@@ -6,10 +6,16 @@ def TARGETS = params.TARGETS ?: "Grinder"
 TARGETS = TARGETS.trim().split("\\s*,\\s*")
 
 def PARALLEL = params.PARALLEL ? params.PARALLEL : "Dynamic"
-def NUM_MACHINES = params.NUM_MACHINES ? params.NUM_MACHINES : 3
+
+def NUM_MACHINES = ""
+if (params.NUM_MACHINES) {
+    NUM_MACHINES = params.NUM_MACHINES
+} else if (!params.TEST_TIME) {
+    NUM_MACHINES = 3
```
Comment (karianna) [inline buildenv/jenkins/aqaTestPipeline.groovy:14]: Can we document this magic number (why 3?)

Comment (llxia) [reply buildenv/jenkins/aqaTestPipeline.groovy:14]: This is a default value if params.NUM_MACHINES is not set

**Comment by smlambert** [comment]
TC pipeline to verify 'old style launch' still works: 
- [AQA_Test_Pipeline/6187/](https://ci.eclipse.org/temurin-compliance/job/AQA_Test_Pipeline/6187/)

**Comment by karianna** [comment]
> * AQA_Test_Pipeline/6187/

It failed...

**Comment by llxia** [comment]
What is the failure? It is job launching related or test case failure? AQA_Test_Pipeline is only for launching the test jobs.

**Comment by karianna** [comment]
The log reads:

Started by remote host 172.30.50.1 with note: Remote triggered by job https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-linux-x64-temurin/266/
Started by user [Shelley Lambert](https://ci.eclipse.org/temurin-compliance/user/slambert@gmail.com)
Rebuilds build #[6161](https://ci.eclipse.org/temurin-compliance/job/AQA_Test_Pipeline/6161)
Running as [Shelley Lambert](https://ci.eclipse.org/temurin-compliance/user/slambert@gmail.com)
Checking out git ${ADOPTOPENJDK_REPO} into /var/jenkins/workspace/AQA_Test_Pipeline@script/6d9b8d1a5fd9730487ba8088635b133c684d197224cb6deede13622845c42fc0 to read aqa-tests/buildenv/jenkins/aqaTestPipeline.groovy
The recommended git tool is: NONE
No credentials specified
 > git rev-parse --resolve-git-dir /var/jenkins/workspace/AQA_Test_Pipeline@script/6d9b8d1a5fd9730487ba8088635b133c684d197224cb6deede13622845c42fc0/aqa-tests/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/llxia/openjdk-tests.git # timeout=10
Pruning obsolete local branches
Cleaning workspace
 > git rev-parse --verify HEAD # timeout=10
Resetting working tree
 > git reset --hard # timeout=10
 > git clean -fdx # timeout=10
Using shallow fetch with depth 1
Fetching upstream changes from https://github.com/llxia/openjdk-tests.git
 > git --version # timeout=10
 > git --version # 'git version 2.45.2'
 > git fetch --tags --force --progress --prune --depth=1 -- https://github.com/llxia/openjdk-tests.git +refs/heads/*:refs/remotes/origin/* # timeout=60
 > git rev-parse origin/mytest^{commit} # timeout=10
Checking out Revision 5d0f06ea8ae73750188f6b63f67c200fc7d12d24 (origin/mytest)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 5d0f06ea8ae73750188f6b63f67c200fc7d12d24 # timeout=10
Commit message: "Improve AQA_Test_Pipeline for release testing"
 > git rev-list --no-walk 5d0f06ea8ae73750188f6b63f67c200fc7d12d24 # timeout=10
[Pipeline] Start of Pipeline
[Pipeline] echo
download_url: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-linux-x64-temurin/266//artifact/workspace/target/OpenJDK21U-jdk_x64_linux_hotspot_21.0.5_3-ea.tar.gz
[Pipeline] echo
TEST_JOB_NAME: Test_openjdk21_hs_sanity.jck_x86-64_linux
[Pipeline] echo
AUTO_AQA_GEN: true
[Pipeline] build (Building Test_Job_Auto_Gen)
Scheduling project: [Test_Job_Auto_Gen](https://ci.eclipse.org/temurin-compliance/job/Test_Job_Auto_Gen/)
Starting building: [Test_Job_Auto_Gen #235](https://ci.eclipse.org/temurin-compliance/job/Test_Job_Auto_Gen/235/)
Build [Test_Job_Auto_Gen #235](https://ci.eclipse.org/temurin-compliance/job/Test_Job_Auto_Gen/235/) completed: SUCCESS
[Pipeline] library
Loading library openjdk-jenkins-helper@master
Examining [adoptium/jenkins-helper](https://github.com/adoptium/jenkins-helper)
Attempting to resolve master as a branch
Resolved master as branch master at revision 3cccb6284f500372ec96c4e58b84130fcc2582cd
The recommended git tool is: NONE
using credential github-bot
 > git rev-parse --resolve-git-dir /var/jenkins/workspace/AQA_Test_Pipeline@libs/34c9ab7e5c7b67e055fc7838253e6124bbe4f3c8945822d9cc219338784b98f3/.git # timeout=10
Fetching changes from the remote Git repository
 > git config remote.origin.url https://github.com/adoptium/jenkins-helper.git # timeout=10
Fetching without tags
Fetching upstream changes from https://github.com/adoptium/jenkins-helper.git
 > git --version # timeout=10
 > git --version # 'git version 2.45.2'
using GIT_ASKPASS to set credentials GitHub bot (username/token)
 > git fetch --no-tags --force --progress -- https://github.com/adoptium/jenkins-helper.git +refs/heads/master:refs/remotes/origin/master # timeout=10
Checking out Revision 3cccb6284f500372ec96c4e58b84130fcc2582cd (master)
 > git config core.sparsecheckout # timeout=10
 > git checkout -f 3cccb6284f500372ec96c4e58b84130fcc2582cd # timeout=10
Commit message: "Add semgrep differential code check action (#61)"
 > git rev-list --no-walk 3cccb6284f500372ec96c4e58b84130fcc2582cd # timeout=10
[Pipeline] echo
TEST_JOB_NAME: Test_openjdk21_hs_special.jck_x86-64_linux
[Pipeline] echo
AUTO_AQA_GEN: true
[Pipeline] build (Building Test_Job_Auto_Gen)
Scheduling project: [Test_Job_Auto_Gen](https://ci.eclipse.org/temurin-compliance/job/Test_Job_Auto_Gen/)
Starting building: [Test_Job_Auto_Gen #236](https://ci.eclipse.org/temurin-compliance/job/Test_Job_Auto_Gen/236/)
Build [Test_Job_Auto_Gen #236](https://ci.eclipse.org/temurin-compliance/job/Test_Job_Auto_Gen/236/) completed: SUCCESS
[Pipeline] library
Only using first definition of library openjdk-jenkins-helper
[Pipeline] parallel
[Pipeline] { (Branch: Test_openjdk21_hs_sanity.jck_x86-64_linux_1)
[Pipeline] { (Branch: Test_openjdk21_hs_special.jck_x86-64_linux_2)
[Pipeline] build (Building Test_openjdk21_hs_sanity.jck_x86-64_linux)
Scheduling project: [Test_openjdk21_hs_sanity.jck_x86-64_linux](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_sanity.jck_x86-64_linux/)
[Pipeline] build (Building Test_openjdk21_hs_special.jck_x86-64_linux)
Scheduling project: [Test_openjdk21_hs_special.jck_x86-64_linux](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_special.jck_x86-64_linux/)
Starting building: [Test_openjdk21_hs_sanity.jck_x86-64_linux #47](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_sanity.jck_x86-64_linux/47/)
Starting building: [Test_openjdk21_hs_special.jck_x86-64_linux #48](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_special.jck_x86-64_linux/48/)
Build [Test_openjdk21_hs_special.jck_x86-64_linux #48](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_special.jck_x86-64_linux/48/) completed: FAILURE
[Pipeline] echo
Test_openjdk21_hs_special.jck_x86-64_linux result is FAILURE
[Pipeline] }
Build [Test_openjdk21_hs_sanity.jck_x86-64_linux #47](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_sanity.jck_x86-64_linux/47/) completed: FAILURE
[Pipeline] echo
Test_openjdk21_hs_sanity.jck_x86-64_linux result is FAILURE
[Pipeline] }
[Pipeline] // parallel
[Pipeline] End of Pipeline
Finished: FAILURE

**Comment by llxia** [comment]
> Build [Test_openjdk21_hs_special.jck_x86-64_linux #48](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_special.jck_x86-64_linux/48/) completed: FAILURE

What is the error message in this build?

**Comment by karianna** [comment]
> > Build [Test_openjdk21_hs_special.jck_x86-64_linux #48](https://ci.eclipse.org/temurin-compliance/job/Test_openjdk21_hs_special.jck_x86-64_linux/48/) completed: FAILURE
> 
> What is the error message in this build?

I must be looking at the wrong thing, I just followed the link from the GH Action

**Comment by smlambert** [comment]
```
19:14:38  + ./get.sh -s /home/jenkins/workspace/Test_openjdk21_hs_special.jck_x86-64_linux/aqa-tests/.. -p x86-64_linux -r customized -j 21 -i hotspot -c https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-linux-x64-temurin/266//artifact/workspace/target/OpenJDK21U-jdk_x64_linux_hotspot_21.0.5_3-ea.tar.gz --clone_openj9 false --tkg_repo https://github.com/adoptium/TKG.git --tkg_branch master
19:14:38  TESTDIR: /home/jenkins/workspace/Test_openjdk21_hs_special.jck_x86-64_linux/aqa-tests
19:14:38  get jdk binary...
19:14:38  _ENCODE_FILE_NEW=UNTAGGED curl -OLJSks --user ****:**** https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-linux-x64-temurin/266//artifact/workspace/target/OpenJDK21U-jdk_x64_linux_hotspot_21.0.5_3-ea.tar.gz
19:14:38  Uncompressing file: OpenJDK21U-jdk_x64_linux_hotspot_21.0.5_3-ea.tar.gz ...
19:14:38  
19:14:38  gzip: OpenJDK21U-jdk_x64_linux_hotspot_21.0.5_3-ea.tar.gz: not in gzip format
19:14:38  tar: This does not look like a tar archive
19:14:38  tar: Exiting with failure status due to previous errors
```

**Comment by llxia** [comment]
The SDK URL is invalid: https://ci.adoptium.net/job/build-scripts/job/jobs/job/jdk21u/job/jdk21u-linux-x64-temurin/266//artifact/workspace/target/OpenJDK21U-jdk_x64_linux_hotspot_21.0.5_3-ea.tar.gz

**Comment by llxia** [comment]
@smlambert could you rerun the pipeline with a valid SDK? Thanks

**Comment by smlambert** [comment]
https://ci.eclipse.org/temurin-compliance/job/AQA_Test_Pipeline/6258/

**Comment by karianna** [comment]
https://ci.eclipse.org/temurin-compliance/job/AQA_Test_Pipeline/6258/console Pipeline ran but there were test failures.

**Comment by llxia** [comment]
Thanks @smlambert !

**Pull request merged**